frontend/test/__support__/server-mocks/collection.ts

```diff
@@ -72,3 +72,97 @@ export function setupUnauthorizedCollectionsEndpoints(
 ) {
   collections.forEach(setupUnauthorizedCollectionEndpoints);
 }
+
+export function setupCollectionByIdEndpoint({
+  collections,
+  error,
+}: {
+  collections: Collection[];
+  error?: string;
+}) {
+  if (error) {
+    setupCollectionWithErrorById({ error });
+    return;
+  }
+
+  fetchMock.get(/api\/collection\/\d+/, url => {
+    const collectionIdParam = url.split("/")[5];
+    const collectionId = Number(collectionIdParam);
+
+    const collection = collections.find(
+      collection => collection.id === collectionId,
+    );
+
+    return collection;
+  });
+}
+
+function setupCollectionWithErrorById({
+  error,
+  status = 500,
+}: {
+  error: string;
+  status?: number;
+}) {
+  fetchMock.get(/api\/collection\/\d+/, () => {
+    return {
+      body: error,
+      status,
+    };
+  });
+}
+
+export function setupRootCollectionItemsEndpoint({
+  collections = [],
+  dashboards = [],
+}: {
+  collections: Collection[];
+  dashboards: Dashboard[];
+}) {
+  fetchMock.get("path:/api/collection/root/items", () => {
+    const rootDashboards = dashboards.filter(
+      dashboard => dashboard.collection_id === null,
+    );
+    const rootCollections = collections.filter(
+      collection => collection.location !== "/",
+    );
+    const data = [...rootDashboards, ...rootCollections];
+
+    return {
+      total: data.length,
+      data,
+    };
+  });
+}
+
+export function setupCollectionItemsEndpoint({
```

Comment:
I think there should be just one helper for both root and regular collection since on the frontend the id of a collection can be "root". I also think that the helper should accept `collection` and `CollectionItem`, like here https://github.com/metabase/metabase/blob/d438e16d03d63f8a457c32ba0ba905d8dfd752a3/frontend/test/__support__/server-mocks/search.ts#L4. If you could fix that with a follow-up PR that would be great